data-reporting-service/observation-service/src/test/java/gov/cdc/etldatapipeline/observation/service/ObservationServiceTest.java

```diff
@@ -75,42 +83,49 @@ private Observation constructObservation(Long observationUid, String obsDomainCd
         Observation observation = new Observation();
         observation.setId(observationUid);
         observation.setObsDomainCdSt1(obsDomainCdSt1);
-        observation.setPersonParticipations(TestUtils.readFileData(filePathPrefix + "PersonParticipations.json"));
+        observation.setPersonParticipations(readFileData(filePathPrefix + "PersonParticipations.json"));
+        observation.setOrganizationParticipations(readFileData(filePathPrefix + "OrganizationParticipations.json"));
+        observation.setMaterialParticipations(readFileData(filePathPrefix + "MaterialParticipations.json"));
+        observation.setFollowupObservations(readFileData(filePathPrefix + "FollowupObservations.json"));
         return observation;
     }
 
-    private void validateData(String inputTopicName, String outputTopicName, String transformedTopicName,
-                              String payload, Long expectedUid) {
+    private void validateData(String inputTopicName, String outputTopicName,
+                              String payload, Observation observation) {
         StreamsBuilder builder = new StreamsBuilder();
-        final var observationService = getObservationService(inputTopicName, outputTopicName, transformedTopicName);
+        final var observationService = getObservationService(inputTopicName, outputTopicName);
         observationService.processMessage(builder);
 
         TopologyTestDriver testDriver = new TopologyTestDriver(builder.build(), new Properties());
         final Serde<String> serdeString = Serdes.String();
         TestInputTopic<String, String> inputTopic = testDriver.createInputTopic(
                 inputTopicName, serdeString.serializer(), serdeString.serializer());
 
-        TestOutputTopic<String, String> outputTopic = testDriver.createOutputTopic(
-                outputTopicName, serdeString.deserializer(), serdeString.deserializer());
-
         inputTopic.pipeInput("100000001", payload);
-        KeyValue<String, String> result = outputTopic.readKeyValue();
-        try {
-            Observation actual = objectMapper.readValue(result.value, Observation.class);
-            assertEquals(expectedUid, actual.getId());
-        } catch (JsonProcessingException e) {
-            e.printStackTrace();
-        }
+
         testDriver.close();
+
+        ObservationKey observationKey = new ObservationKey();
+        observationKey.setObservationUid(observation.getId());
+
+        ObservationReporting reportingModel = modelMapper.map(observation, ObservationReporting.class);
+        final ObservationTransformed transformed = transformer.transformObservationData(observation);
```

Comment:
Do we explictly need to do transformation here. Won't the TopolgyTestDriver call the kafkaStream processing automatically.

Reply:
Yes, it does. But unfortunately TopologyDriver can't initialize output topic if the service does not end with KStream.to(outputTopic...) like we have in this kind of services (kafkaTemplate used here instead). So there is no way to compare output value from the output topic (like in person-service), and test compares expected transformed model with the value that is send via kafkaTemplate...

Reply:
yes, this transformation is to process that nested jsons coming from the stored proc